### ISSUE TITLE
Is Microsoft.Extensions.Logging 3.0.0-preview.19074.2 necessary?

### ISSUE THREAD

**Issue by malachib** [Half-Shot/matrix-dotnet-sdk#7]
It doesn't seem to be, and is stopping me from compiling a netcore2.2 app

**Comment by Half-Shot** [comment]
Yes sure! I will probably switch to a better logger (probs `Serilog`). I'm going to be a little busy for a while yet, so if you want to go ahead and remove the dependency (I don't think I make extensive use of the logger) and PR it then that would be cool :)

**Comment by malachib** [comment]
I instead encourage to further embrace Microsoft's logger interface - I'm sure we can make it gracefully live in your environment !  If I have time to do a PR definitely will.  I tried removing the dependency, but need to fiddle with the factory more.  

Also, separately, hit an issue with Client.Keys, wasn't present so wouldnt compile -- any guidance there?

**Comment by Half-Shot** [comment]
Yeah, it was missing some stuff, commited those.
